Tests/SystemTests/FilePathTests/FilePathDecodable.swift

```diff
@@ -0,0 +1,127 @@
+/*
+ This source file is part of the Swift System open source project
+ 
+ Copyright (c)2024 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See https://swift.org/LICENSE.txt for license information
+ */
+
+import XCTest
+
+#if SYSTEM_PACKAGE
+@testable import SystemPackage
+#else
+@testable import System
+#endif
+
+@available(/*System 0.0.1: macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0*/iOS 8, *)
+final class FilePathDecodableTest: XCTestCase {
+  func testInvalidFilePath() {
+    // _storage is a valid SystemString, but the invariants of FilePath are
+    // violated.
```

Comment:
Which invariant? Is there any way to have a textual representation?

Reply:
The textual representation for these is not very helpful, since the SystemString is basically garbage (hence unnormalized, which is the invariant that fails). We should add additional tests for the other invariants, but these came from examples provided by Johannes in a bug report.

Reply:
(Just FWIW, when I noticed something was off, I wrote a little (super basic) fuzzer based on just randomness. I then extracted these examples, I don't actually know what they hit.)